### DISCUSSION TITLE
How to compare enum which type is from another package?

### DISCUSSION THREAD

**Discussion by piratf** [antonmedv/expr#236] [Q&A]
For example: 
``` golang
import (
    someapi "xxxx"
)

type SomeStruct struct {
    Type someapi.SomeType	
}
```

The expression I want to execute is to validate the value of an enumeration.
```
someStruct.Type == someapi.SomeType_TypeOne
```

errors encountered:
failed: unknown name someapi
This error is easy to understand, but I don't know how to import the "someapi" package into `expr` context.

I tried to convert the type to int32, but it didn't work
```
int32(someStruct.Type.Number())==1
```
failed: unknown func int32

The only solution I've found so far is to convert to string
```
someStruct.Type.String()=="Day"
```

This looks more expensive, so I want to seek the best practice.

**Comment by antonmedv** [comment]
You can use Patch to override this part.

**Comment by antonmedv** [comment]
In order to use someapi it should be a struct. 

Unfortunately Go’s reflect doesn’t support reflection on packages.
